pkg/operators/kubenetworkmanager/kubenetworkmanager.go

```diff
@@ -0,0 +1,322 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubenetworkmanager
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/k8sutil"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/kubemanager"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+type SetIPToK8sResolverInterface interface {
+	SetIPToK8sResolver(resolver func(ip string) string)
+}
+
+// TODO: Where should this be defined?
+type RemoteKind string
+
+const (
+	RemoteKindPod     RemoteKind = "pod"
+	RemoteKindService RemoteKind = "svc"
+	RemoteKindOther   RemoteKind = "other"
+)
+
+// TODO: More granualar and "shareable" interfaces
+type AdditionalPodNetworkInformation interface {
+	// Local/Host
+	SetPodOwner(string)
+	SetPodHostIP(string)
+	SetPodIP(string)
+	SetPodLabels(map[string]string)
+
+	// Remote
+	GetRemoteIP() string
+	SetRemoteNamespace(string)
+	SetRemoteKind(string)
+	SetRemotePodLabels(map[string]string)
+}
+
+// TODO: Generalize this. Will be useful for other gadgets/operators too
+type k8sInventoryCache struct {
+	clientset *kubernetes.Clientset
+
+	pods atomic.Pointer[v1.PodList]
+	svcs atomic.Pointer[v1.ServiceList]
+
+	exit   chan bool
+	ticker *time.Ticker
+}
+
+func newCache(tickerDuration time.Duration) (*k8sInventoryCache, error) {
+	clientset, err := k8sutil.NewClientset("")
+	if err != nil {
+		return nil, fmt.Errorf("creating new k8s clientset: %w", err)
+	}
+
+	cache := &k8sInventoryCache{
+		clientset: clientset,
+		exit:      make(chan bool),
+		ticker:    time.NewTicker(tickerDuration),
+	}
+
+	// Store empty lists
+	cache.pods.Store(&v1.PodList{})
+	cache.svcs.Store(&v1.ServiceList{})
+
```

Comment:
Maybe `cache.update()` once here to make sure that even the first event will get the information?